server/src/main/java/no/ntnu/trostespel/udpServer/GameDataSender.java

```diff
@@ -41,15 +41,10 @@ public GameDataSender() {
     /**
      * Does the update tasks for the server.
      */
-    public void broadcast(List<Connection> connections, long tick) throws InterruptedException {
-        /*if (completedCount.get() != 0) {
-            throw new InterruptedException("Game data broadcast was interrupted");
-            // TODO: Change approach; this can potentially deadlock if one connection hangs
-        }*/
+    public void broadcast(List<Connection> connections, long tick) {
         // Send GameState to all clients
         nextGameState = gameStateMaster.getGameState();
         nextGameState.setTick(tick);
-        // TODO: 15.10.2019 Add concurrency protection, since we will be modifying connecitons on the fly.
         String json = gson.toJson(nextGameState, RECEIVED_DATA_TYPE);
         connectionsSize = connections.size();
         for (Connection con : connections) {
```

Comment:
Should check if connection is conencted or disconnected before sending?

Reply:
I'm on it